src/pyodide-py/pyodide/webloop.py

```diff
@@ -0,0 +1,425 @@
+            self._next_handle = h
+
+        if self._next_handle is not None and self._next_handle._when <= self.time():
+            h = self._next_handle
+            self._next_handle = None
+            self._immediate.append(h)
+
+        not_finished = (
+            self._immediate or self._scheduled or self._next_handle or self._futures
+        )
+        if forever or (until_complete and not_finished):
+            self._timeout_promise(self._interval).then(
+                lambda x: self._do_tasks(until_complete=until_complete, forever=forever)
+            )
+        else:
+            self._quit_running()
+
+    def _quit_running(self):
+        """
+        Quit running
+        """
+        if asyncio.get_event_loop() == self:
+            asyncio._set_running_loop(None)
+        self._running = False
+        if self._stop_callbacks:
+            resolve, reject = self._stop_callbacks
+            self._stop_callbacks = None
+            if self._exception:
+                reject(self._exception)
+            else:
+                resolve(self._result)
+
+    def _timer_handle_cancelled(self, handle: Callable):
+        """
+        Timer handle canlled
+        """
+        pass
+
+    def is_running(self) -> bool:
+        """
+        Return True if the event loop is currently running.
+        """
+        return self._running
+
+    def is_closed(self) -> bool:
+        """
+        Return True if the event loop was closed.
+        """
+        return not self._running
+
+    def stop(self):
+        """
+        Stop the event loop.
+        """
+        self._stop = True
+        self._quit_running()
+
+    def close(self):
+        """
+        Close the event loop.
+        """
+        self._stop = True
+        self._quit_running()
+
+    def shutdown_asyncgens(self):
+        raise NotImplementedError
+
+    def shutdown_default_executor(self):
+        raise NotImplementedError
+
+    def default_exception_handler(self, context: Dict) -> Callable:
+        """
+        Default exception handler.
+        """
+        js.console.error(context.get("message"))
+
+    def call_exception_handler(self, context: Dict):
+        """
+        Call the current event loop exception handler.
+
+        Parameters
+        ----------
+        context
+          context is a dict object containing the following keys (new keys may be introduced in future Python versions):
+            ‘message’: Error message;
+            ‘exception’ (optional): Exception object;
+            ‘future’ (optional): asyncio.Future instance;
+            ‘handle’ (optional): asyncio.Handle instance;
+            ‘protocol’ (optional): Protocol instance;
+            ‘transport’ (optional): Transport instance;
+            ‘socket’ (optional): socket.socket instance.
+        """
+        if self._exception_handler:
+            self._exception_handler(self, context)
+        else:
+            self.default_exception_handler(context)
```

Comment:
```suggestion
        self._exception_handler(self, context)
```

Reply:
I implemented as you suggested in my last version but then I noticed that it was wrong according to the docs. The issue is a custom exception handler takes two arguments(loop, context) while the default handler takes one argument (context), so we cannot really treat them the same.

Reply:
Add a comment please?

Reply:
Added a comment to note the difference.